Packs/Cisco-umbrella/Integrations/CiscoUmbrellaEnforcement/CiscoUmbrellaEnforcement.py

```diff
@@ -0,0 +1,197 @@
+        suffix += f'limit={limit}&'
+    return suffix
+
+
+def domains_list_command(client: Client, args: dict) -> CommandResults:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param args: args from the user for the command.
+    """
+    page = args.get('page', '')
+    limit = args.get('limit', '')
+    response = client.get_domains_list(page=page, limit=limit)
+    readable_output = tableToMarkdown(t=response, name='List of Domains', headers=['id', 'name'])
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix='CiscoUmbrellaEnforcement.Domains',
+        outputs_key_field='id',
+        outputs=response
+    )
+
+
+def domain_event_add_command(client: Client, args: dict) -> str:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param args: args from the user for the command.
+    :returns (str) confirmation or error regarding adding a new domain.
+    """
+    alert_time = args.get('alert_time')
+    device_id = args.get('device_id')
+    dst_domain = args.get('destination_domain')
+    dst_url = args.get('destination_url')
+    device_version = args.get('device_version')
+
+    new_event = {
+        "alertTime": alert_time,
+        "deviceId": device_id,
+        "deviceVersion": device_version,
+        "dstDomain": dst_domain,
+        "dstUrl": dst_url,
+        "eventTime": alert_time,
+        "protocolVersion": "1.0a",
+        "providerName": "Security Platform"
+    }
+
+    response: dict = client.add_event_to_domain(new_event)
+    if id := str(response.get('id')):
+        action_result = f"New event was added successfully, The id is {id}."
+    else:
+        action_result = "New event's addition failed."
+    return action_result
+
+
+def domain_delete_command(client: Client, args: dict) -> CommandResults:
+    """
+    :param client: Cisco Umbrella Client for the api request.
+    :param args: args from the user for the command.
+    :returns (str) confirmation or error regarding deleting a domain.
+    """
+    response = {}
+    domain_name = args.get('name', '')
+    domain_id = args.get('id', '')
+    if not domain_name and not domain_id:
+        raise DemistoException(
+            'Both domain name and domain id do not exist, Please supply one of them in order to set the domain to '
+            'delete command')
+    try:
+        response = client.delete_domains(domain_id=domain_id, domain_name=domain_name)
+    except DemistoException as e:
+        if e.res.get('statusCode') == 404:
+            return CommandResults(
+                readable_output='The domain was not found in the list, Please insert an existing domain name or id.'
+            )
+    if domain_name:
+        old_context = demisto.dt(demisto.context(), f'CiscoUmbrellaEnforcement.Domains(val.name == "{domain_name}")')
+    else:
+        old_context = demisto.dt(demisto.context(), f'CiscoUmbrellaEnforcement.Domains(val.id == "{domain_id}")')
```

Comment:
didn't we want to do it once with `||`?

Reply:
Yes however it did not work for me, as always there will be only one given arg or id or name.
when one of the args is empty the dt search will look :`'CiscoUmbrellaEnforcement.Domains(val.id == || val.name="test.com")'`. this search does not work and the only way that it does is inserting quotes inside the string which I straggled with. If I'll find a way to make it work I'll change it.